### PR TITLE
Uses jsonschema API

### PR DESCRIPTION
**Pull request by tkohr**
... from https://github.com/camptocamp/drealcorse-reports/pull/26 without filtering schemas by layers as schema does not yet include the `layer_id`.

### CODE REVIEW

File: js/extension/components/FeatureReports.jsx [hunk 1]
```diff
@@ -18,14 +27,41 @@ class FeatureReports extends React.Component {
...
+        postReport(formData);
+    }
+
+    
+
     render() {
         const { selectedSchema, editReport } = this.state;
 
+        console.log(this.props);
+
```
Comment (tonio) [inline js/extension/components/FeatureReports.jsx]:
```suggestion
```

File: js/extension/state/actions.js [hunk 2]
```diff
@@ -30,3 +30,19 @@ export const selectSchema = (payload) => {
         payload
     };
 };
+
+export const postReport = (payload) => {
+    console.log("post report action")
```
Comment (tonio) [inline js/extension/state/actions.js]:
```suggestion
```

File: js/extension/state/epics.js [hunk 3]
```diff
@@ -68,4 +69,53 @@ export const displayFormEpic = (action$, store) => action$.ofType('DISPLAY_FORM'
...
+    console.log("post report epic, data : ", data);
+
+    // use axios in prod, fetch in dev
+    const postAPI = process.env.NODE_ENV === 'production' ?
+    httpReportsProd.post('', data,)
+        .then(response => response.data) 
+        .then(response => console.log(response))
+        .then(data => {
+            console.log('Success:', data);
+        })
```
Comment (tonio) [inline js/extension/state/epics.js]: too much log (-:
in this promise chain, `console.log()` returns `undefined`, then `data` of line 90 will be undefined.
what you returns in the `then` of a promise is the argument of the next `then()` in the chain.

File: js/extension/components/FeatureReports.jsx [hunk 4]
```diff
@@ -18,14 +27,41 @@ class FeatureReports extends React.Component {
...
+    featureToString(feature) {
+        return JSON.stringify({feature}, null, "4");
+    }
+
+    schemaOption(schema) {
+        return {label: schema.name, value: schema};
+    }
+
+    selectSchema(schema) {
+        schema.formData ? schema.formData.feature_id = '' : schema.formData = {"feature_id" : ''}
```
Comment (tonio) [inline js/extension/components/FeatureReports.jsx]: I would go back to a good old `if` statement. Not easily readable that way.

File: js/extension/state/epics.js [hunk 5]
```diff
@@ -1,37 +1,38 @@
...
-    //     fetch('https://georchestra.mydomain.org/mapstore-reports/reports')
-    //         .then(response => response.json());
-
-    // return Rx.Observable.fromPromise(fetchAPI)
-    //     .switchMap((response) => {
-    //         console.log('Fetched schemas: ', response)
-    //         return Rx.Observable.of(loadedSchemas(response))
-    //     })
-    //     .catch(e => Rx.Observable.of(loadError(e.message)));
+    // return Rx.Observable.of(loadedSchemas(mockSchemas));
```
Comment (tonio) [inline js/extension/state/epics.js:19]: we may think to a way of being able to use mock/or local compo. But this it out of this PR scope.

File: js/extension/state/epics.js [hunk 6]
```diff
@@ -1,37 +1,38 @@
...
+    // TODO: replace 'reports' API (used to test ping) by 'schemas' API once it exists
+    const fetchAPI = process.env.NODE_ENV === 'production' ?
+        axios.get('../mapstore-reports/jsonschemas')
+            .then(response => response.data) :
+        fetch('https://georchestra.mydomain.org/mapstore-reports/jsonschemas')
+            .then(response => response.json());
+
+    return Rx.Observable.fromPromise(fetchAPI)
+        .switchMap((response) => {
+            console.log('Fetched schemas: ', response);
```
Comment (tonio) [inline js/extension/state/epics.js]:
```suggestion
```

File: js/extension/state/epics.js [hunk 7]
```diff
@@ -1,37 +1,38 @@
...
+            const responseList = Object.values(response);
+            return Rx.Observable.of(loadedSchemas(responseList));
```
Comment (tonio) [inline js/extension/state/epics.js:32]: I wondered if this could only be:

```suggestion
            return loadedSchemas(responseList);
```

**Review by tonio — COMMENTED** [review]
👍🏻 Nice work, it may require a bit of cleaning before going in.

File: js/extension/state/epics.js (hunk 7 above)
Comment (isanchez-c2c) [reply js/extension/state/epics.js:32]: No, it doesn't work.

**Pull request merged**